projects/plugins/jetpack/tests/e2e/specs/post-connection/sync.test.js

```diff
@@ -0,0 +1,106 @@
+import { test, expect } from '../../fixtures/base-test.js';
+import { execWpCommand } from 'jetpack-e2e-commons/helpers/utils-helper.cjs';
+import { BlockEditorPage } from 'jetpack-e2e-commons/pages/wp-admin/index.js';
+import { prerequisitesBuilder } from 'jetpack-e2e-commons/env/index.js';
+import playwrightConfig from '../../playwright.config.cjs';
+
+test.describe( 'Sync', () => {
```

Comment:
@fgiannar  I'm happy to see new tests being added. 🥇 

Here are my thoughts:

As an e2e test, I think your approach is good. 

I would probably also add a check that sync type is actually enabled. Both Normal Sync and Dedicated Sync tests are checking that sync works, but they don't check how it works. If enabling dedicated sync fails, I assume normal sync will work and the test will pass, but it will be a false positive because it will actually test something else.

On testing via the UI vs API:

While posting via the API will be faster (and most probably more stable) it wouldn't be an e2e test, it wouldn't be testing the real user flow. 
But, in this context, does it matter how the post is created? Is the sync process affected in any way if the post is made via the UI or via the API? If not, then testing it via a UI e2e test is indeed not necessary.
On the other hand, I wouldn't add an API test in this suite. That test would belong to a different layer (integration), with a different setup, one that we're not currently having. I'm happy this came up because maybe we should start considering it. Most probably many flows like this would be better tested like that.

To conclude, I think the implemented approach is good for us now at this moment, and I'll take the need of testing the API layer on me.

Reply:
> Is the sync process affected in any way if the post is made via the UI or via the API? If not, then testing it via a UI e2e test is indeed not necessary.
> On the other hand, I wouldn't add an API test in this suite. That test would belong to a different layer (integration), with a different setup, one that we're not currently having. I'm happy this came up because maybe we should start considering it. Most probably many flows like this would be better tested like that.

Specifically for Sync we don't care how the post was added. It's pretty agnostic if the post was added via the UI or API, as we want to sync all the data over, no matter how it was added. Testing fully via API/integration tests would be a bit better in the long run as it would be much faster to run the tests in the various configurations and less error prone on UI changes that happen from time to time. 

For Sync specifically it would be highly beneficial if we can get such framework in place to be able to tests in such a flow in the future. 

Most of our tests should be contained to running a few API calls to the WordPress/Jetpack site to generate test data and adjust settings and then running a few calls to the WordPress.com API to verify the results were successful or vice-versa. No need for fancy UI (as most of the under the hood settings don't have or won't need a UI on the Jetpack site and will be controllable through WordPress.com's backend UIs).

Reply:
> For Sync specifically it would be highly beneficial if we can get such framework in place to be able to tests in such a flow in the future.

I will take this in our check-in meetings and see what makes more sense for us.